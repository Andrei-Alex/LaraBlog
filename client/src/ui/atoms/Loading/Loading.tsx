import { styles } from '.';

export default function Loading() {
  return (
    <div className={styles.container}>
      <p>Loading...</p>
    </div>
  );
}
