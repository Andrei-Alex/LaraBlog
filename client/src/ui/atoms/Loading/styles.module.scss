@keyframes opacityChange {
     100% {
        opacity: 1;
    }
    50% {
        opacity: 0.1;
    }
}

.container {
    @apply text-center my-10;
    p {
        @apply text-xl text-gray-500 font-semibold;
        animation: opacityChange 2s infinite;
    }
}
