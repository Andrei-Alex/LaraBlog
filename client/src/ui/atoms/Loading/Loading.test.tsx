import { render, screen } from '@testing-library/react';

import Loading from './Loading';
import { styles } from '.';

describe('Loading', () => {
  it('renders correctly and matches snapshot', () => {
    const { asFragment } = render(<Loading />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders and applies correct classes', () => {
    render(<Loading />);
    const loadingElement = screen.getByText(/Loading.../i);
    expect(loadingElement.parentElement).toHaveClass(styles.container);
  });
});
