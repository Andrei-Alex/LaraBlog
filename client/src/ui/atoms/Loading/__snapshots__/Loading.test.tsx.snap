// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <p>
      Loading...
    </p>
  </div>
</DocumentFragment>
`;
